chart/templates/metrics-service.yaml

```diff
@@ -0,0 +1,21 @@
+{{- if .Values.metrics.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: {{ template "argo-tunnel.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    {{ .Values.metrics.servicemonitor.label }}: {{ .Values.metrics.servicemonitor.value }}
+  name: {{ template "argo-tunnel.fullname" . }}-metrics
```

Comment:
The service is the piece that the prometheus servicemonitor will discover and begin scraping.

Reply:
interesting! I usually configure prometheus for both pod and service scraping (e.g. allow discovery directly from the pod annotation).  It probably because i'm a caveman (not using the prometheus operator).

If we are adding a service any way, what are your feelings on always having a service definition, called `{{ template "argo-tunnel.fullname" . }}`, then its just add the label if you enable scraping?